tests/raft_scenarios/retire

```diff
@@ -0,0 +1,50 @@
+start_node,0
+assert_is_primary,0
+emit_signature,2
+
+assert_is_primary,0
+assert_commit_idx,0,2
+
+trust_node,2,1
+emit_signature,2
+
+dispatch_all
+periodic_all,10
+dispatch_all
+assert_commit_idx,0,4
+# Initial network state with nodes 0 and 1
+
+# Reconfig [0, 1] -> [0, 2]
+swap_node,2,0,2
+emit_signature,2
+
+# Leader replicates
+periodic_one,0,10
+dispatch_one,0
+
+# But node 1 runs for election before ack/cmt
```

Comment:
There may be distinct cases if we do this _before_ and _after_ the ACK is received.  In particular, after the ACKs arrive at the primary (but before anything further has been shared), the primary can advance commit, and get into their final retirement state.

Reply:
Agreed, but I think that's the same/maybe worse, definitely no better.

Reply:
In fact, this is a red herring caused by my mistaken thought that the primary counting itself in vs rejecting AE attempts from other candidates would matter. But that is not the case.

What matters is whether all the possible primaries post-retirement have a way to advance commit, and this is no more true whether the initial primary steps down gracefully or is interrupted at the moment.